CMake/external_libcurl.cmake

```diff
@@ -0,0 +1,33 @@
+if(ENABLE_RS_AUTO_UPDATER)
+    include(ExternalProject)
+    message(STATUS "Building libcurl enabled")
+    
+    set(CURL_FLAGS -DBUILD_CURL_EXE=OFF -DBUILD_SHARED_LIBS=OFF -DUSE_WIN32_LDAP=OFF -DHTTP_ONLY=ON -DCURL_ZLIB=OFF -DCURL_DISABLE_CRYPTO_AUTH=ON -DCMAKE_USE_OPENSSL=OFF -DCMAKE_USE_LIBSSH2=OFF)
+    if (WIN32)
+        set(CURL_FLAGS ${CURL_FLAGS} -DCURL_STATIC_CRT=ON )
+    endif()
+
+    ExternalProject_Add(
+        libcurl
+        PREFIX libcurl
+        GIT_REPOSITORY "https://github.com/curl/curl.git"
+        GIT_TAG "53cdc2c963e33bc0cc1a51ad2df79396202e07f8" # curl-7_70_0
+        SOURCE_DIR ${CMAKE_CURRENT_BINARY_DIR}/third-party/libcurl
```

Comment:
Do we really want it in third-party/? Conceptually, this is a temporary project downloaded by CMake and therefore should be cleaned with the cache?

What does 'git status' report for all stuff in third-party after this is cloned? Is it ignored?

Reply:
Note that the source directory is under "CMAKE_CURRENT_BINARY_DIR"
This means it will be under the "build" library which is under the .gitIgnore list.
I place it there parallel to the "libusb" example that also imported on build using ExternalProject_add.

Reply:
Gotcha. Good.